Author: Jaret Cantu <jaret.cantu@timesys.com>
Date:   Sun Mar 22 15:24:45 EDT 2015

    The capability of MySQL to use various threading libraries instead of
    only NPTL has been removed for years, but this one rarely occurring
    reference to a removed global variable and constant remained:

    /.../mysql-5.6.23/mysys/stacktrace.c:358:27: error: 'thd_lib_detected' undeclared (first use in this function)
    sigreturn_frame_count = thd_lib_detected == THD_LIB_LT ? 2 : 1;
                            ^
    /.../mysql-5.6.23/mysys/stacktrace.c:358:27: note: each undeclared identifier is reported only once for each function it appears in
    /.../mysys/stacktrace.c:358:47: error: 'THD_LIB_LT' undeclared (first use in this function)
    sigreturn_frame_count = thd_lib_detected == THD_LIB_LT ? 2 : 1;
                                                ^

    It was only encountered on x86 uClibc builds.
    This code has been removed in the 5.7 development branch.

--- mysql-5.6.23/mysys/stacktrace.c.orig	2015-03-22 15:21:00.746848789 -0400
+++ mysql-5.6.23/mysys/stacktrace.c	2015-03-22 15:22:22.626843553 -0400
@@ -291,8 +291,11 @@ inline uint32* find_prev_pc(uint32* pc,
 
 void my_print_stacktrace(uchar* stack_bottom, ulong thread_stack)
 {
+  /* We are 1 frame above signal frame with NPTL */
+  const uint sigreturn_frame_count = 1;
   uchar** fp;
-  uint frame_count = 0, sigreturn_frame_count;
+  uint frame_count = 0;
+
 #if defined(__alpha__) && defined(__GNUC__)
   uint32* pc;
 #endif
@@ -354,9 +357,6 @@ void my_print_stacktrace(uchar* stack_bo
 		      :"r"(pc));
 #endif  /* __alpha__ */
 
-  /* We are 1 frame above signal frame with NPTL and 2 frames above with LT */
-  sigreturn_frame_count = thd_lib_detected == THD_LIB_LT ? 2 : 1;
-
   while (fp < (uchar**) stack_bottom)
   {
 #if defined(__i386__) || defined(__x86_64__)
